subsys/tracing/CMakeLists.txt

```diff
@@ -27,10 +27,14 @@ zephyr_sources_ifdef(
   tracing_backend_uart.c
   )
 
-zephyr_sources_ifdef(
-  CONFIG_TRACING_BACKEND_POSIX
-  tracing_backend_posix.c
-  )
+if (CONFIG_TRACING_BACKEND_POSIX)
+  zephyr_sources(tracing_backend_posix.c)
+  if (CONFIG_NATIVE_APPLICATION)
```

Comment:
Just to understand, when are we not `CONFIG_NATIVE_APPLICATION`?

Reply:
When building for the native simulator (native_sim and native_sim_64 so far).
These claim CONFIG_NATIVE_LIBRARY (which later is built into an app)